bfvmm/include/hve/arch/intel_x64/vcpu.h

```diff
@@ -186,6 +163,39 @@ class EXPORT_HVE vcpu : public bfvmm::vcpu
     ///
     VIRTUAL bool advance();
 
+
+    /// Initialize vCPU
+    ///
+    /// Initializes the vCPU.
+    ///
+    /// @expects none
+    /// @ensures none
+    ///
+    /// @return returns true if successful, false otherwise
+    ///
+    bool init();
+
+    /// De-initialize vCPU
+    ///
+    /// De-initializes the vCPU.
+    ///
+    /// @expects none
+    /// @ensures none
+    ///
+    /// @return returns true if successful, false otherwise
+    ///
+    bool fini();
+
+    /// Add exit handler
+    ///
+    /// Adds an exit handler to be called on every vmexit
+    ///
+    /// @expects none
+    /// @ensures none
+    ///
+    void add_exit_handler(const handler_delegate_t &d)
```

Comment:
This **is** the function you call to add an exit handler for every exit reason. I think I renamed it because there used to be two different functions for adding exit handlers, we can name it whatever if its confusing now.